gson/src/org/immutables/gson/stream/JsonParserReader.java

```diff
@@ -143,7 +144,12 @@ public String nextString() throws IOException {
   @Override
   public boolean nextBoolean() throws IOException {
     requirePeek();
-    boolean value = parser.getBooleanValue();
+    boolean value;
+    try {
+      value = parser.getBooleanValue();
```

Comment:
Can you use `parser.getCurrentToken()` to detect JSON type ?

Reply:
Also what is the default behavior of GSON `JsonReader.nextBoolean()`:

```json
{ 
  "foo": true, 
  "bar": "true"
} 
```

will it return `true` in both cases ?

Reply:
This is not an issue for Gson or the Jackson JsonParser. They both correctly distinguish the token type as boolean. The issue is with the Jackson XML JsonParser. All boolean tokens come come across as a string tokens. So calling `JsonReader.nextBoolean()` Fails here https://github.com/FasterXML/jackson-core/blob/2.11/src/main/java/com/fasterxml/jackson/core/JsonParser.java#L1426-L1431

Reply:
`JsonReader.nextBoolean()` should only be called when `currentToken` is `VALUE_TRUE` or  `VALUE_FALSE`. 

Perhaps something is wrong elsewhere (like method `getCurrentToken()`) ?